api/src/main/java/org/apache/iceberg/transforms/Timestamps.java

```diff
@@ -53,7 +53,8 @@ public Integer apply(Long timestampMicros) {
     OffsetDateTime timestamp = Instant
         .ofEpochSecond(timestampMicros / 1_000_000)
         .atOffset(ZoneOffset.UTC);
-    return (int) granularity.between(EPOCH, timestamp);
+    Integer year = Long.valueOf(granularity.between(EPOCH, timestamp)).intValue();
```

Comment:
This isn't necessarily a year. It may be months, days, or hours. Can we return `intValue()` directly instead?